packages/node-core/src/db/db.module.ts

```diff
@@ -39,17 +40,44 @@ async function establishConnection(sequelize: Sequelize, numRetries: number): Pr
   }
 }
 
-const sequelizeFactory = (option: SequelizeOption, migrate: boolean) => async () => {
-  const sequelize = new Sequelize(option);
-  const numRetries = 5;
-  await establishConnection(sequelize, numRetries);
-  await sequelize.sync({alter: migrate});
-  return sequelize;
-};
+const sequelizeFactory =
+  (option: SequelizeOption, project: ISubqueryProject<IProjectNetworkConfig>, nodeConfig: NodeConfig) => async () => {
+    const sequelize = new Sequelize(option);
+    const numRetries = 5;
+    await establishConnection(sequelize, numRetries);
+
+    const {chainId} = project.network;
+    const {dbSchema, migrate} = nodeConfig;
+
+    let chainIdMatch = false;
+
+    if (migrate) {
+      try {
+        const result = await sequelize.query<{chain_id: string}>(
+          `SELECT chain_id FROM "${dbSchema}"._metadata ORDER BY "createdAt"`,
+          {type: QueryTypes.SELECT}
+        );
+        if (result[0].chain_id === chainId) chainIdMatch = true;
```

Comment:
rather than compare with chainId from project.yaml (as it is editable), we should compare it with existing genesishHash/chainId in metadata